packages/web-components/fast-foundation/src/data-grid/data-grid-cell.ts

```diff
@@ -0,0 +1,304 @@
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import {
+    keyCodeEnter,
+    keyCodeEscape,
+    keyCodeFunction2,
+} from "@microsoft/fast-web-utilities";
+import { ColumnDefinition } from "./data-grid";
+
+const defaultCellContentsTemplate: ViewTemplate = html<DataGridCell>`
+    <template>
+        ${x =>
+            x.rowData === null ||
+            x.columnDefinition === null ||
+            x.columnDefinition.columnDataKey === null
+                ? null
+                : x.rowData[x.columnDefinition.columnDataKey]}
+    </template>
+`;
+
+const defaultHeaderCellContentsTemplate: ViewTemplate = html<DataGridCell>`
+    <template>
+        ${x =>
+            x.columnDefinition === null
+                ? null
+                : x.columnDefinition.title === undefined
+                ? x.columnDefinition.columnDataKey
+                : x.columnDefinition.title}
+    </template>
+`;
+
+/**
+ * Enumerates possible cell types.
+ *
+ * @public
+ */
+export enum DataGridCellTypes {
+    default = "default",
+    columnHeader = "columnheader",
+}
+
+/**
+ * A Data Grid Cell Custom HTML Element.
+ *
+ * @public
+ */
+export class DataGridCell extends FASTElement {
+    /**
+     * The type of cell
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: cell-type
+     */
+    @attr({ attribute: "cell-type" })
+    public cellType: DataGridCellTypes;
+    private cellTypeChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.updateCellView();
+        }
+    }
+
+    /**
+     * The column index of the cell.
+     * This will be applied to the css grid-column-index value
+     * applied to the cell
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: grid-column
+     */
+    @attr({ attribute: "grid-column" })
+    public gridColumn: string;
+    private gridColumnChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.updateCellStyle();
+        }
+    }
+
+    /**
+     * The base data for the parent row
+     *
+     * @public
+     */
+    @observable
+    public rowData: object | null = null;
+
+    /**
+     * The base data for the column
+     *
+     * @public
+     */
+    @observable
+    public columnDefinition: ColumnDefinition | null = null;
+    private columnDefinitionChanged(
+        oldValue: ColumnDefinition | null,
+        newValue: ColumnDefinition | null
+    ): void {
+        if (this.$fastController.isConnected) {
+            this.updateCellView();
+        }
+    }
+
+    private isActiveCell: boolean = false;
+    private customCellView: HTMLView | null = null;
+    private isInternalFocused: boolean = false;
+
+    /**
+     * @internal
+     */
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        this.addEventListener("focusin", this.handleFocusin);
+        this.addEventListener("focusout", this.handleFocusout);
+        this.addEventListener("keydown", this.handleKeydown);
```

Comment:
As a follow up, consider referencing the `@microsoft/fast-web-utilities` package, I have just created a PR that exports these event type strings https://github.com/microsoft/fast/pull/4161

Reply:
Done.